common/src/main/java/io/netty/util/internal/PlatformDependent.java

```diff
@@ -150,33 +167,16 @@ public Random current() {
             };
         }
 
-        // Here is how the system property is used:
-        //
-        // * <  0  - Don't use cleaner, and inherit max direct memory from java. In this case the
-        //           "practical max direct memory" would be 2 * max memory as defined by the JDK.
-        // * == 0  - Use cleaner, Netty will not enforce max memory, and instead will defer to JDK.
-        // * >  0  - Don't use cleaner. This will limit Netty's total direct memory
-        //           (note: that JDK's direct memory limit is independent of this).
-        long maxDirectMemory = SystemPropertyUtil.getLong("io.netty.maxDirectMemory", -1);
-
-        if (maxDirectMemory == 0 || !hasUnsafe() || !PlatformDependent0.hasDirectBufferNoCleanerConstructor()) {
-            USE_DIRECT_BUFFER_NO_CLEANER = false;
-            DIRECT_MEMORY_COUNTER = null;
-        } else {
-            USE_DIRECT_BUFFER_NO_CLEANER = true;
-            if (maxDirectMemory < 0) {
-                maxDirectMemory = MAX_DIRECT_MEMORY;
-                if (maxDirectMemory <= 0) {
-                    DIRECT_MEMORY_COUNTER = null;
-                } else {
-                    DIRECT_MEMORY_COUNTER = new AtomicLong();
-                }
-            } else {
-                DIRECT_MEMORY_COUNTER = new AtomicLong();
-            }
+        /*
+         * We do not want to log this message if unsafe is explicitly disabled. Do not remove the explicit no unsafe
+         * guard.
+         */
+        if (!hasUnsafe() && !isAndroid() && !PlatformDependent0.isExplicitNoUnsafe()) {
+            logger.info(
+              "Your platform does not provide complete low-level API for accessing direct buffers reliably. " +
+                "Unless explicitly requested, heap buffer will always be preferred to avoid potential system " +
+                "instability.");
         }
```

Comment:
This "if" block should be removed.

From the versions we forked, It was present in versions of netty up to 4.1.27. https://github.com/netty/netty/blob/1c16519ac86d9ef9919ac96afbee4be9b428d0c9/common/src/main/java/io/netty/util/internal/PlatformDependent.java#L139
Then from the version 4.1.48 this was removed and replaced by another block (same message but different checks). https://github.com/netty/netty/blob/c10c697e5bf664d9d8d1dcee93569265b19ca03a/common/src/main/java/io/netty/util/internal/PlatformDependent.java#L211

Reply:
Good catch thanks, done https://github.com/sirensolutions/netty/pull/29/commits/ec4cb36b3e44b2916621024f498094ae101f3691